### ISSUE TITLE
[DatePicker] Input value is cleared when closing the datepicker

### ISSUE THREAD

**Issue by igor-serzhan** [mui/mui-x#4478]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Input value is cleared after open and close the datepicker dialog.

### Expected behavior 🤔

Input value should stay the same

### Steps to reproduce 🕹

Steps:

1. Open https://codesandbox.io/s/k860y9?file=/demo.js (official demo)
2. Type valid date in input (for example 03/03/2022)
3. Blur the input
4. Open Datepicker Dialog (click at the calendar icon)
5. Close Datepicker Dialog (just click outside the dialog or press 'Esc' button)
6. See that input value is empty

### Context 🔦

https://codesandbox.io/s/k860y9?file=/demo.js

### Your environment 🌎

environment from the official sandbox

**Comment by flaviendelangle** [comment]
Yes, I am working on it on #4408 and @yyq321 did some work as well on #4383

I would be interested in other opinion about those two comments

https://github.com/mui/mui-x/pull/4383#issuecomment-1093019932
https://github.com/mui/mui-x/pull/4383#issuecomment-1095016396
